stdlib/ballerina-http/src/main/java/org/ballerinalang/net/uri/URIUtil.java

```diff
@@ -62,4 +68,52 @@ public static void populateQueryParamMap(String queryParamString, BMap<String, B
             }
         }
     }
+
+    public static BMap<String, BValue> getMatrixParamsMap(String path, HTTPCarbonMessage carbonMessage) {
+        BMap<String, BValue> matrixParamsBMap = new BMap<>();
+        Map<String, Map<String, String>> pathToMatrixParamMap =
+                (Map<String, Map<String, String>>) carbonMessage.getProperty(Constants.MATRIX_PARAMS);
+        if (!pathToMatrixParamMap.containsKey(path)) {
+            return matrixParamsBMap;
```

Comment:
Shall we avoid this return statement by putting the below code to a if check with `pathToMatrixParamMap.containsKey(path)` and have one return statement at the end. IMO This will result in more readable code